tests/versioned/apollo-federation/transaction-inline-fragments.test.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const { executeQuery } = require('../../test-client')
+
+const { setupFederatedGatewayServerTests } = require('./federated-gateway-server-setup')
+const { checkResult } = require('../common')
+
+setupFederatedGatewayServerTests({
+  suiteName: 'query with inline fragments',
+  createTests: createFederatedSegmentsTests,
+  instrumentSubGraphs: true
+})
+
+/**
+ * Creates a federated server that instruments all sub graphs and makes a
+ * query that will contain InlineFragments to a subgraph
+ * It is required that t.context.helper and t.context.serverUrl are set.
+ * @param {*} t a tap test instance
+ */
```

Comment:
An example of a breaking subgraph query could be cool in the comment here. Super minor so will approve but if you want to add may help us in the future if we forget what that looks like.

Reply:
i can add that, don't merge this yet